src/IPGenerate/IPGenerator.cpp

```diff
@@ -464,6 +468,9 @@ bool IPGenerator::Generate() {
           m_compiler->ErrorMessage("IP Generate, " + help.str());
           return false;
         }
+
+        // Mark this instance as generated
+        inst->setGenerated(true);
```

Comment:
This is not correct, what if the parameters have changed, then it needs to be re-generated.
What you need is to have a Generate(ip_instance)  which takes a parameter to generate the IP under generation in the GUI.
And for the ipgenerate command, there is already (double check) a logic that geneates incrementaly the the parameters have not changed, if that is not working fix it.
But this high level status as you propose it is not correct.

Reply:
Hey Alain, could you check this logic I added awhile back and see if it addresses your concern? https://github.com/skyler-rs/FOEDAG/blob/d886c3e2b9ae775c3c150381dd7ff86b41497823/src/IPGenerate/IPGenerator.cpp#L246
At some point i had to add logic for to replace an existing instance if a user reconfigured, w/ the same module name so any re-call off configure_ip will replace the existing instance which will then get generated on next ipgenerate and the instance will then be marked as generated.

If this approach is not appropriate, could you point me to the logic you mentioned? I don't see it in IPGenerator::Generate()

Reply:
I've refactored this PR to pass a module name as you suggested. I copied the SynthMoreOpts approach to pass the modules name(s) and added a List type to IPGenerateOpt to handle the scenario. Let me know if you were expected a different method for passing the module name.